src/GraphQLParser/ParserContext.Parse.cs

```diff
@@ -25,6 +26,9 @@ public GraphQLDocument ParseDocument()
             );
             _document.Definitions = definitions;
             _document.UnattachedComments = _unattachedComments;
+
+            Debug.Assert(_currentDepth == 1, "Depth has not returned to 1 after parsing document");
```

Comment:
This is not compiled within release builds, but will reliable fail tests if the number of `IncreaseDepth()` and `DecreaseDepth()` calls are mismatched.

Reply:
👍